example/matrixAddWithMdspan/src/matrixAddMdSpan.cpp

```diff
@@ -0,0 +1,166 @@
+/* Copyright 2023 Mehmet Yusufoglu, Bernhard Manfred Gruber, René Widera, Andrea Bocci, Simeon Ehrig
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+// Needed for running example for all backends available; one by one
+#include <alpaka/example/ExecuteForEachAccTag.hpp>
+
+#include <experimental/mdspan>
+#include <iostream>
+
+/**
+ * @brief Kernel for performing addition of two 2D matrices. Each element is added by a different thread. MdSpan data
+ * structure is used to pass the data to and from the kernel.
+ */
+struct MatrixAddKernel
+{
+    //! \tparam TAcc Accelerator type
+    //! \tparam MdSpan The type of the multidimensional span (mdspan)
+    //! \param acc Accelerator
+    //! \param A First input matrix
+    //! \param B Second input matrix
+    //! \param C Output matrix where the result of A + B will be stored
+    template<typename TAcc, typename MdSpan>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc, MdSpan A, MdSpan B, MdSpan C) const
```

Comment:
One thing I don't like here is that there is no check that `MdSpan` is actually an `mdspan` object, nor that it has 2 dimensions.

Reply:
But I'm fine keeping it undefined for the time being, if we plan to restrict it with concepts once we move to c++20.

Reply:
I have added a type check. Hopefully users will not find it complicated.

Reply:
You can also easily check the dimension with the static function `rank()`.

```c++
        static_assert(MdSpan::rank() == 2);
```

Reply:
done, thanks.